README.md

```diff
@@ -1,151 +1,145 @@
-(bash): validate -t file-input path/to/file.bam
-```
+| File Type | Description |
+| :-------: | ------ |
+| bam | Validate bam/cram/sam using `pysam`. <br> Check for an index file in same directory as the BAM.<br><br>_Note: If a BAM input is missing an accompanying BAM index file in the same directory,<br> `validate` will not throw an exception but will print a warning._|
+| vcf | Validate vcf using `vcftools` |
+| fasta |  |
+| bed | |
+| py | |
 
-_Running as Nextflow process with docker_
-```
-check the example under /example/ or the pipeline-align-DNA repository
-```
+_Note: If the input is invalid in any way, `validate` will exit with a non-zero status code._
 
-### usage commands:
-#### file specific validation
-Currently file type specific validation is supported for the following:
-| type | tool |
-|------|------|
-| bam | pysam |
-| vcf | vcftools |
 
-To explicitly check a single file type, run
+### Expected Output
+
+Valid input
 ```
-validate -t file-py path/to/file.py
+Input: path/to/input is valid
 ```
-Where file-py can be replaced with any file type listed in the input types table.
-To automatically detect any or multiple file types, run
+Invalid input or error
 ```
-validate -t file-input path/to/file.ext
+Error: path/to/input Error Message
 ```
-The tool will try to automatically detect the file type and do file specific validation, and if the file type is unsupported, will just do a simple existence check.
 
-#### directory specific validation
-To run validation for checking basic directory permissions you can run the following
+### How To Run
 
+**Parameters**
+
+Required args:<br>
+- _`path`_ 
+   - path of one or more files to validate
+
+Optional args
+- _`-h`, `--help`_ 
+   - show the help message and exit
+
+#### File Validation
+
+To automatically detect any or multiple file types, run
 ```
-validate -t directory-rw path/to/directory/
-```
-```
-validate -t directory-r path/to/directory/
+validate path/to/file.ext
 ```
 
-Using "directory-r" for read permissions, and "directory-rw" for read and write permissions.
+## Generate Checksum
+Generate a new checksum file based on the input file path. Or generates checksum comparison if `.md5` or `.sha512` file exists.
+
+### How To Run
+
+**Parameters**
+
+Required arg
+- _path_ path of one or more files or directories to validate
+
+Optional args
+- _-t, --type_ checksum type
+- _-h, --help_ show this help message and exit
 
-#### checksum generation (beta)
-To generate a sha512 or md5 checksum file using this tool, run the following
 
+#### Checksum Generation (beta)
+Generate a checksum file for the specified file in the same directory. Available types: `md5`, `sha512`
+
+for `md5`:
 ```
 generate-checksum -t md5 path/to/file.ext
 ```
 
+for `sha512`:
 ```
 generate-checksum -t sha512 path/to/file.ext
 ```
 
-Available types: `md5`, `sha512`
+## References
 
-It should create a checksum file at the path/to/file.ext.{checksum_ext}
+### Pysam
```

Comment:
wasn't sure if it would be better to separate references by tool if it made it easier later for some reason. I can just put them in a repo and publication section together for all dependencies if that's preferred.  Or just not label them as repo or publication at all.

Reply:
I'm ok with the way they are now!